types/lib/model.d.ts

```diff
@@ -648,6 +646,57 @@ export interface Silent {
   silent?: boolean;
 }
 
+type FunctionKeys<T> = ({[P in keyof T]: T[P] extends Function ? P : never })[keyof T];
+type ArrayType<T> = NonNullable<T> extends Array<infer R> ? R : any;
```

Comment:
I'm not quite sure where to put these helper types. The custom `Omit` type was defined in `model.d.ts` as well, but it doesn't feel like the right place.

Reply:
I think it's fine to leave them here for now, we can worry about refactoring on another time